geometry/rp2_element_body.hpp

```diff
@@ -0,0 +1,90 @@
+
+#pragma once
+
+#include "geometry/rp2_element.hpp"
+
+#include <cmath>
+#include <string>
+
+#include "quantities/quantities.hpp"
+
+namespace principia {
+namespace geometry {
+namespace internal_rp2_element {
+
+using quantities::DebugString;
+using quantities::Infinity;
+
+template<typename Scalar>
+RP2Element<Scalar>::RP2Element(
+    Scalar const& x, Scalar const& y, double z)
+    : x_(x), y_(y), z_(z) {
+  // [0:0:0] does not represent any point.
+  CHECK(x != Scalar() || y != Scalar() || z != 0.0);
```

Comment:
I am skeptical; [underflow:underflow:underflow] should probably not cause an outright failure...